interpreter/interpreter_test.go

```diff
@@ -667,3 +686,43 @@ func getSideEffectsValues(ses []interpreter.SideEffect) []values.Value {
 	}
 	return vs
 }
+
+func TestStack(t *testing.T) {
+	src := `from(bucket: "telegraf") |> range(start: -5m) |> aggregateWindow(every: 1m, fn: mean)`
+	ctx := dependenciestest.Default().Inject(context.Background())
+	sideEffects, _, err := runtime.Eval(ctx, src)
+	if err != nil {
+		t.Fatal(err)
+	} else if len(sideEffects) != 1 {
+		t.Fatalf("expected 1 side effect, got %d", len(sideEffects))
+	}
+
+	to, ok := sideEffects[0].Value.(*flux.TableObject)
+	if !ok {
+		t.Fatal("expected side effect to be a table object")
+	}
+
+	want := []interpreter.StackEntry{
+		{
+			FunctionName: "window",
+			Location: ast.SourceLocation{
+				File:   "universe.flux",
+				Start:  ast.Position{Line: 119, Column: 12},
+				End:    ast.Position{Line: 119, Column: 49},
+				Source: `window(every:inf, timeColumn:timeDst)`,
+			},
+		},
+		{
+			FunctionName: "aggregateWindow",
+			Location: ast.SourceLocation{
+				Start:  ast.Position{Line: 1, Column: 50},
+				End:    ast.Position{Line: 1, Column: 86},
+				Source: `aggregateWindow(every: 1m, fn: mean)`,
+			},
+		},
+	}
```

Comment:
Why do only `window` and `aggregateWindow` show up in the stack?

Reply:
The aggregate window call invokes window as part of the function body. The previous transformations are piped arguments so they are not part of the stack trace. Is there another stack trace you would expect?